### PR TITLE
feat: Add way to control the close/open functionality via API

### PR DESCRIPTION
**Pull request by umbobabo**
In this PR https://github.com/EconomistDigitalSolutions/fe-blogs/pull/1647 I need to control the open/close functionality of the ballon with a state from redux. So here I'm adding a prop that makes the component controllable from the outside.

Prettify add some additional formatting on top of the PR.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -77,37 +92,45 @@ class Balloon extends React.Component {
   changeVisibility(visibility) {
     const { dynamicPositioning, unstyled } = this.props;
     if (!visibility) {
-      visibility = (this.state.visibility === 'not-visible') ? 'visible' : 'not-visible';
+      visibility = this.state.visibility === 'not-visible' ? 'visible' : 'not-visible';
```
Comment (rinkashimikito) [inline src/index.js]: It's just my personal preference but I would keep those parenthesis. They are adding more readability in code and they will be removed by babel.

Comment (rinkashimikito) [inline src/index.js]: It's just my personal preference but I would keep those parenthesis. They are adding more readability in code and they will be removed by babel.

File: src/index.js [hunk 2]
```diff
@@ -25,20 +25,35 @@ class Balloon extends React.Component {
...
     showOnHoverDelay: 100,
-  }
+    visible: false,
+  };
 
   constructor(props) {
     super(props);
     this.state = {
-      visibility: 'not-visible',
+      visibility: this.props.visible ? 'visible' : 'not-visible',
```
Comment (rinkashimikito) [inline src/index.js]: Do we need to set `this.state.visibility` as a `string` ? Can't we just use boolean values?

File: src/index.js (hunk 1 above)
Comment (rinkashimikito) [reply src/index.js]: That includes other similar situations

File: src/index.js (hunk 2 above)
Comment (umbobabo) [reply src/index.js]: We use it as a classname. Anyway it was like that already so I would like to not spend time refactoring if not necessary.

File: src/index.js (hunk 1 above)
Comment (umbobabo) [reply src/index.js]: That has been done automatically by prettier plugin via my IDE. I think we can postpone this discussion and put in relation to the Prettier story we have in the backlog. I will revert this change.

File: src/index.js (hunk 2 above)
Comment (rinkashimikito) [reply src/index.js]: Fair enough 👍

File: src/index.js (hunk 1 above)
Comment (rinkashimikito) [reply src/index.js]: Ok cool

**Comment by rinkashimikito** [comment]
LGTM :shipit:

**Pull request merged**